src/controllers/exporter/ExporterController.php

```diff
@@ -0,0 +1,99 @@
+<?hh // strict
+
+abstract class ExporterController {
+  abstract public function genData(string $type): Awaitable<mixed>;
+
+  public async function genTeams(): Awaitable<mixed> {
+    $all_teams_data = array();
+    $all_teams = \HH\Asio\join(Team::genAllTeams());
+
+    foreach ($all_teams as $team) {
+      $team_data = \HH\Asio\join(Team::genTeamData($team->getId()));
+      $one_team = array(
+        'name' => $team->getName(),
+        'active' => $team->getActive(),
+        'admin' => $team->getAdmin(),
+        'protected' => $team->getProtected(),
+        'visible' => $team->getVisible(),
+        'password_hash' => $team->getPasswordHash(),
+        'points' => $team->getPoints(),
+        'logo' => $team->getLogo(),
+        'data' => $team_data
+      );
+      array_push($all_teams_data, $one_team);
+    }
+    return $all_teams_data;
+  }
+
+  public async function genLevels(): Awaitable<mixed> {
+    $all_levels_data = array();
+    $all_levels = \HH\Asio\join(Level::genAllLevels());
+
+    foreach ($all_levels as $level) {
+      $entity = \HH\Asio\join(Country::gen($level->getEntityId()));
```

Comment:
One of the main benefits of await is you can do things like `await gena(array(gen1, gen2))`, and gen1 and gen2 will be awaited in parallel. He's saying `gen` all the entity IDs in one command. In any case, no new code should ever use `join`.